### ISSUE TITLE
search-engine: Wrong type argument: commandp, ("Amazon" . engine/search-amazon)

### ISSUE THREAD

**Issue by r-darwish** [syl20bnr/spacemacs#6542]
<<REPLACE ALL UPPERCASE EXPRESSIONS AND DELETE THIS LINE, `C-C C-C` TO SUBMIT>>
#### Description :octocat:
spacemacs/search-engine-select doesn't work

#### Reproduction guide :beetle:
- Start Emacs
- Hit SPC a /
- Select any search engine

*Observed behaviour:* :eyes: :broken_heart:
Wrong type argument: commandp, ("Amazon" . engine/search-amazon)

*Expected behaviour:* :heart: :smile:
Get prompted for the search term

#### System Info :computer:
- OS: darwin
- Emacs: 24.5.1
- Spacemacs: 0.105.21
- Spacemacs branch: develop (rev. f1247c8)
- Graphic display: t
- Distribution: spacemacs
- Editing style: hybrid
- Completion: ivy
- Layers:
```elisp
(osx darwish ansible
     (auto-completion :variables auto-completion-private-snippets-directory dcp-snippet-dir auto-completion-complete-with-key-sequence-delay 0.0 auto-completion-enable-snippets-in-popup t)
     emacs-lisp
     (go :variables go-tab-width 4)
     restclient docker search-engine ivy python javascript html git org github markdown semantic rust c-c   systemd
     (spell-checking :variables spell-checking-enable-by-default nil)
     (shell :variables shell-default-shell 'eshell)
     colors
     (syntax-checking :variables syntax-checking-enable-tooltips nil)
     (version-control :variables version-control-diff-tool 'diff-hl))
```

#### Backtrace :paw_prints:
```
Debugger entered--Lisp error: (wrong-type-argument commandp ("Amazon" . engine/search-amazon))
  call-interactively(("Amazon" . engine/search-amazon))
  (lambda (candidate) (call-interactively candidate))(("Amazon" . engine/search-amazon))
  ivy-call()
  ivy-read("Search Engines: " (("Amazon" . engine/search-amazon) ("Bing" . engine/search-bing) ("Duck Duck Go" . engine/search-duck-duck-go) ("Google" . engine/search-google) ("Google Images" . engine/search-google-images) ("Github" . engine/search-github) ("Google Maps" . engine/search-google-maps) ("Twitter" . engine/search-twitter) ("Project Gutenberg" . engine/search-project-gutenberg) ("YouTube" . engine/search-youtube) ("Stack Overflow" . engine/search-stack-overflow) ("Spacemacs Issues" . engine/search-spacemacs-issues) ("Spacemacs Pull Requests" . engine/search-spacemacs-pullrequests) ("Wikipedia" . engine/search-wikipedia) ("Wolfram Alpha" . engine/search-wolfram-alpha)) :action (lambda (candidate) (call-interactively candidate
```

**Comment by JorisE** [comment]
I'm not able to reproduce this. It sounds like a package mixup.
I'm guessing that removing the elpa directory from .emacs.d and restarting emacs will fix this. All packages (correct versions) will be reinstalled.

**Comment by CestDiego** [comment]
it might be that the ivy backend is being wrong here.

**Comment by d12frosted** [comment]
Thanks to @nixmaniack this is fixed in develop.

**Comment by r-darwish** [comment]
Seems to work now. Thanks.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
